src/components/AssetDetailComponent.jsx

```diff
@@ -1,37 +1,28 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import _, { isEmpty, values } from 'lodash';
+import { isEmpty } from 'lodash';
 import { Container, Header } from 'semantic-ui-react';
 import { getAssetDetail, allocateAsset, unassignAsset } from '../_actions/asset.actions';
 import { loadAssetAssigneeUsers } from '../_actions/users.actions';
 import AssetDetailContent from './AssetDetailContent';
 import NavbarComponent from './NavBarComponent';
+import LoaderComponent from './LoaderComponent';
 
 export class AssetDetailComponent extends Component {
   state = {
-    assignedUser: {},
-    selectedUser: 0,
     serialNumber: '',
-    assignAssetButtonState: true,
     hasError: this.props.hasError,
     errorMessage: this.props.errorMessage
   };
 
   componentDidMount() {
-    this.getAssetId(this.props.location.pathname);
-    if (_.isEmpty(this.props.assetAsigneeUsers)) {
-      this.props.loadAssetAssigneeUsers();
+    const { assetDetail } = this.props;
+    if (isEmpty(assetDetail)) {
+      this.getAssetId(this.props.location.pathname);
```

Comment:
I think you should rename the method to `getAssetDetails` as you are retrieving the details not id

Reply:
it gets asset id from the path 
`const stringArray = pathName.split('/');`
 `const serialNumber = stringArray[2];`

Reply:
@Thuku  Ohh...ok. In thought it was fetching the details. Maybe rename it to `getIdFromPathname`